### ISSUE TITLE
"Discrete value supplied to continuous scale" error when ggplot2/ggmosaic is imported

### ISSUE THREAD

**Issue by kweinert** [haleyjeppson/ggmosaic#58]
I am having trouble using ggmosaic from within a package. First, I thought this is due to my very basic ggplot2 knowledge and [asked on stackoverflow](https://stackoverflow.com/questions/70447288/ggmosaic-discrete-value-supplied-to-continuous-scale-error-when-ggplot2-ggmos). But maybe it is a problem that cannot be fixed from the outside?

**Comment by GegznaV** [comment]
I added a reprex to illustrate this issue:

``` r
library(tidyverse)

test_ggmosaic_1 <- function() {
  ggplot(datasets::mtcars) +
    ggmosaic::geom_mosaic(aes(x = ggmosaic::product(gear), fill = cyl)) +
    scale_fill_discrete()
}


## ERROR when `ggmosaic::` is used
test_ggmosaic_1()
#> Error: Discrete value supplied to continuous scale


# OK when package is fully loaded
library(ggmosaic)
test_ggmosaic_1()
```

![](https://i.imgur.com/MmUHb2d.png)

<sup>Created on 2021-12-23 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

I expected the same output in both cases above, but the first one results in an error.

**Comment by kellijohnson-NOAA** [comment]
A fix is given in #36 that I can confirm worked for me in [{nwfscSurvey}](www.github.com/pfmc-assessmets/nwfscSurvey) where I added `ggmosaic::scale_x_productlist()` to the last line of the {ggplot2} object.
